patterns/template-home-business.php

```diff
@@ -1,7 +1,7 @@
 <?php
 /**
  * Title: Business home template
- * Slug: twentytwentyfour/template-home
+ * Slug: twentytwentyfour/template-home-business
  * Template Types: front-page, index, home
```

Comment:
Let's remove `index` here, so that the home business template is not suggested as a recommended index template (it's not suited well as an index template).

Reply:
![CleanShot 2023-10-30 at 11 08 00](https://github.com/WordPress/twentytwentyfour/assets/1813435/1de785f7-1194-4d63-b7f3-8a0c436649b4)

Reply:
I've done this in a separate PR here: https://github.com/WordPress/twentytwentyfour/pull/709